File: woodroid/src/main/java/me/gilo/woodroid/models/filters/ListFilter.java
package me.gilo.woodroid.models.filters;

import me.gilo.woodroid.utils.Converter;

import java.util.*;

public class ListFilter {

    String context;
    int page;
    int per_page;
    String search;
    String after;
    String before;
    int[] exclude;
    int[] include;
    int offset;
    String order;
    String orderby;


    Map<String, String> filters = new HashMap<>();

    public String getContext() {
        return context;
    }

    public void setContext(String context) {
        this.context = context;

        addFilter("context", context);
    }

    public int getPage() {
        return page;
    }

    public void setPage(int page) {
        this.page = page;

        addFilter("page", page);
    }

    public int getPer_page() {
        return per_page;
    }

    public void setPer_page(int per_page) {
        this.per_page = per_page;

        addFilter("per_page", per_page);
    }

    public String getSearch() {
        return search;
    }

    public void setSearch(String search) {
        this.search = search;

        addFilter("search", search);
    }

    public String getAfter() {
        return after;
    }

    public void setAfter(String after) {
        this.after = after;

        addFilter("after", after);
    }

    public String getBefore() {
        return before;
    }

    public void setAfter(Date date) {
        setAfter(Converter.getDateString(date));
    }

    public void setBefore(Date date) {
        setBefore(Converter.getDateString(date));
    }

    public void setBefore(String before) {
        this.before = before;

        addFilter("before", before);
    }

    public int[] getExclude() {
        return exclude;
    }

    public void setExclude(int[] exclude) {
        this.exclude = exclude;

        addFilter("exclude", exclude);
    }

    public int[] getInclude() {
        return include;
    }

    public void setInclude(int[] include) {
        this.include = include;

        addFilter("include", include);
    }

    public int getOffset() {
        return offset;
    }

    public void setOffset(int offset) {
        this.offset = offset;

        addFilter("offset", offset);
    }

    public String getOrder() {
        return order;
    }

    public void setOrder(String order) {
        this.order = order;

        addFilter("order", order);
    }

    public void setOrder(Sort order) {
       setOrder(order.toString());
    }

    public String getOrderby() {
        return orderby;
    }

    public void setOrderby(String orderby) {
        this.orderby = orderby;

        addFilter("orderby", orderby);
    }

    public void addFilter(String filter, Object value){
        if (value.getClass().isArray()){
            if (value instanceof int[]) {
                String values = "";
                int[] items = (int[]) value;
                for (int item : items){
                    values += item + ",";
                }

                filters.put(filter, ""+values.substring(0, values.length() - 1));

            }else {
                List list = Arrays.asList(value);
                for (Object item : list) {
                    filters.put(filter, item.toString());
                }
            }
        }else {
            filters.put(filter, value.toString());
        }
    }

    public Map<String, String> getFilters() {
        return filters;
    }
    
}

File: woodroid/src/main/java/me/gilo/woodroid/models/filters/ListFilter.kt
package me.gilo.woodroid.models.filters

import me.gilo.woodroid.utils.Converter

import java.util.*

open class ListFilter {

    internal lateinit var context: String
    internal var page: Int = 0
    internal var per_page: Int = 0
    internal lateinit var search: String
    internal lateinit var after: String
    internal lateinit var before: String
    internal lateinit var exclude: IntArray
    internal lateinit var include: IntArray
    internal var offset: Int = 0
    internal lateinit var order: String
    internal lateinit var orderby: String


    internal var filters: MutableMap<String, String> = HashMap()

    fun getContext(): String {
        return context
    }

    fun setContext(context: String) {
        this.context = context

        addFilter("context", context)
    }

    fun getPage(): Int {
        return page
    }

    fun setPage(page: Int) {
        this.page = page

        addFilter("page", page)
    }

    fun getPer_page(): Int {
        return per_page
    }

    fun setPer_page(per_page: Int) {
        this.per_page = per_page

        addFilter("per_page", per_page)
    }

    fun getSearch(): String {
        return search
    }

    fun setSearch(search: String) {
        this.search = search

        addFilter("search", search)
    }

    fun getAfter(): String {
        return after
    }

    fun setAfter(after: String) {
        this.after = after

        addFilter("after", after)
    }

    fun getBefore(): String {
        return before
    }

    fun setAfter(date: Date) {
        setAfter(Converter.getDateString(date))
    }

    fun setBefore(date: Date) {
        setBefore(Converter.getDateString(date))
    }

    fun setBefore(before: String) {
        this.before = before

        addFilter("before", before)
    }

    fun getExclude(): IntArray {
        return exclude
    }

    fun setExclude(exclude: IntArray) {
        this.exclude = exclude

        addFilter("exclude", exclude)
    }

    fun getInclude(): IntArray {
        return include
    }

    fun setInclude(include: IntArray) {
        this.include = include

        addFilter("include", include)
    }

    fun getOffset(): Int {
        return offset
    }

    fun setOffset(offset: Int) {
        this.offset = offset

        addFilter("offset", offset)
    }

    fun getOrder(): String {
        return order
    }

    fun setOrder(order: String) {
        this.order = order

        addFilter("order", order)
    }

    fun setOrder(order: Sort) {
        setOrder(order.toString())
    }

    fun getOrderby(): String {
        return orderby
    }

    fun setOrderby(orderby: String) {
        this.orderby = orderby

        addFilter("orderby", orderby)
    }

    fun addFilter(filter: String, value: Any) {
        if (value.javaClass.isArray) {
            if (value is IntArray) {
                var values = ""
                for (item in value) {
                    values += "$item,"
                }

                filters[filter] = "" + values.substring(0, values.length - 1)

            } else {
                val list = Arrays.asList(value)
                for (item in list) {
                    filters[filter] = item.toString()
                }
            }
        } else {
            filters[filter] = value.toString()
        }
    }

    fun getFilters(): Map<String, String> {
        return filters
    }

}
